internal/services/hdinsight/hdinsight_cluster_pool_resource.go

```diff
@@ -0,0 +1,464 @@
+			if err != nil {
+				return err
+			}
+
+			if err = client.ClusterPoolsDeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandClusterPoolProfile(profiles []ClusterPoolProfile) *hdinsights.ClusterPoolProfile {
+	if len(profiles) == 0 {
+		return nil
+	}
+
+	result := hdinsights.ClusterPoolProfile{
+		ClusterPoolVersion: profiles[0].ClusterPoolVersion,
+	}
+	return &result
+}
+
+func expandComputeProfile(profiles []ComputeProfile) hdinsights.ClusterPoolComputeProfile {
+	result := hdinsights.ClusterPoolComputeProfile{
+		VMSize: profiles[0].VmSize,
+	}
+	return result
+}
+
+func expandLogAnalyticsProfile(profiles []LogAnalyticsProfile) *hdinsights.ClusterPoolLogAnalyticsProfile {
+	if len(profiles) == 0 {
+		return nil
+	}
+
+	result := hdinsights.ClusterPoolLogAnalyticsProfile{
+		Enabled:     profiles[0].LogAnalyticsProfileEnabled,
+		WorkspaceId: pointer.To(profiles[0].WorkspaceId),
+	}
+	return &result
+}
+
+func expandNetworkProfile(profiles []NetworkProfile) *hdinsights.ClusterPoolNetworkProfile {
+	if len(profiles) == 0 {
+		return nil
+	}
+
+	result := hdinsights.ClusterPoolNetworkProfile{
+		SubnetId: profiles[0].SubnetId,
+	}
+
+	if profiles[0].PrivateApiServerEnabled {
+		result.EnablePrivateApiServer = pointer.To(true)
+	}
+
+	if profiles[0].OutboundType != "" {
+		result.OutboundType = pointer.To(hdinsights.OutboundType(profiles[0].OutboundType))
+	}
+	return &result
+}
+
+func flattenClusterPoolProfile(input *hdinsights.ClusterPoolProfile) []ClusterPoolProfile {
+	result := make([]ClusterPoolProfile, 0)
+	if input == nil {
+		return result
+	}
+
+	profile := ClusterPoolProfile{
+		ClusterPoolVersion: input.ClusterPoolVersion,
+	}
+	result = append(result, profile)
+
+	return result
+}
+
+func flattenComputeProfile(input hdinsights.ClusterPoolComputeProfile) []ComputeProfile {
+	result := make([]ComputeProfile, 0)
+	profile := ComputeProfile{
+		VmSize: input.VMSize,
+	}
+	result = append(result, profile)
+
+	return result
+}
+
+func flattenLogAnalyticsProfile(input *hdinsights.ClusterPoolLogAnalyticsProfile) []LogAnalyticsProfile {
+	result := make([]LogAnalyticsProfile, 0)
+	if input == nil {
+		return result
+	}
+
+	profile := LogAnalyticsProfile{
+		LogAnalyticsProfileEnabled: input.Enabled,
+		WorkspaceId:                pointer.From(input.WorkspaceId),
+	}
+	result = append(result, profile)
+
+	return result
+}
+
+func flattenNetworkProfile(input *hdinsights.ClusterPoolNetworkProfile) []NetworkProfile {
+	result := make([]NetworkProfile, 0)
+	if input == nil {
+		return result
+	}
+
+	profile := NetworkProfile{
+		SubnetId: input.SubnetId,
+	}
+
+	if input.EnablePrivateApiServer != nil {
+		profile.PrivateApiServerEnabled = *input.EnablePrivateApiServer
+	}
+
+	if input.OutboundType != nil {
+		profile.OutboundType = string(*input.OutboundType)
+	}
+	result = append(result, profile)
+
+	return result
+}
```

Comment:
I am confused, why are all of these fields lists when all of the expand and flatten functions are not looping over multiple values, they are only looking at the 0 index?

Reply:
Shouldn't we flatten these code blocks then?

Reply:
As requested by service team POC, I think we'd better keep them this way?